src/analyses/is_threaded.cpp

```diff
@@ -33,10 +33,8 @@ class is_threaded_domaint:public ai_domain_baset
     locationt from,
     locationt to)
   {
-    // assert(src.reachable);
-
-    if(!src.reachable)
-      return false;
+    INVARIANT(src.reachable,
```

Comment:
Probably `PRECONDITION` here expresses the intent clearer.

Reply:
@thk123 I'm unsure... my belief is that this is enforced by a property of the AI engine; it's really an invariant of that.  Hmmm... maybe it is a precondition...

Reply:
It's both a precondition of the method and a property of the AI engine, thus I used the more general `INVARIANT`. However, thinking about it now maybe `PRECONDITION` does indeed suit better, and that the AI engine does not apply `transform()` to bottom should be checked by an invariant in `ait` instead.